pkg/admissioncontroller/apis/config/types.go

```diff
@@ -35,6 +35,9 @@ type AdmissionControllerConfiguration struct {
 	Server ServerConfiguration
 	// Debugging holds configuration for Debugging related features.
 	Debugging *componentbaseconfig.DebuggingConfiguration
+	// AllowTokenFile specifies if "tokenFile" field is allowed to be used in the kubeconfig file
+	// used for connecting to the Garden cluster.
+	AllowTokenFile *bool
```

Comment:
I would even say, that this should not be a field in the config API.
Instead of plumbing this flag all the way through to the client creation, WDYT about generally allowing token files during client creation, except when creating plant clients? This was the original motivation behind forbidding the `*File` fields in kubeconfigs (@rfranzke do I remember correctly?).
Considering that more and more use cases for token files are coming up these days, I invert the logic and generally allow them to support all the rising use cases.

Reply:
Good point, I think it's safe to allow `tokenFile`s for at least garden, seed or shoot client creations. For plants we need to indeed be cautious since the kubeconfig can be provided by end-users.

Reply:
Is this addressed?

Reply:
I am not sure what do you mean by this. I removed the `allowTokenFile` configuration and explicitly allowed the `admission-controller` to use `tokenFile` field in its kubeconfig.

Maybe your suggesting was to allow `tokenFile` everywhere and then forbid it only for the plants? On the other hand a lot of components (e.g. `gardenlet`) will be allowed something that is not needed for them. I guess it is low to none risk since the `kubeconfig`s cannot come from "the outside", but why allow a functionality that is not going to be used?

Reply:
> Maybe your suggestion was to allow `tokenFile` everywhere and then forbid it only for the plants?

Yes, that was exactly my suggestion.

> but why allow a functionality that is not going to be used?

You could also ask: why add a restriction that is not needed? 😉 

So anyways, thanks for explaining your motivation. We can go ahead like this :)